src/apps/pillarx-app/components/WalletPortfolioButtons/WalletPortfolioButtons.tsx

```diff
@@ -38,23 +44,157 @@ const WalletPortfolioButtons = () => {
     !isPrivyConnectedViaWalletConnect &&
     kit.getEtherspotProvider().getWalletMode() !== 'delegatedEoa';
 
+  const handleAddCash = async () => {
+    if (isAddCashLoading) return;
```

Comment:
Can we show something to the user here so they're not ghosted - just an alert telling them that it's still loading and to try again shortly